src/multi-select/multi-select-item/multi-select-item.component.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { PropTypes } from 'react';
+import { Checkbox } from 'react-bootstrap';
+
+import './multi-select-item.component.scss';
+
+export default class MultiSelectItem extends React.PureComponent {
+
+  static propTypes = {
+    item: PropTypes.shape({
+      id: PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+      ]).isRequired,
+      text: PropTypes.string.isRequired,
+    }).isRequired,
+    isChecked: PropTypes.bool.isRequired,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    onChange: () => {},
+  };
+
+  shouldComponentUpdate(nextProps) {
```

Comment:
This should be obsolete when item is passed directly without using object literal from parent component. When item is passed directly then the shallow comparison of PureComponent works.